specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-05-01-preview/machineLearningServices.json

```diff
@@ -4996,6 +5703,320 @@
+                    "Key"
+                  ],
+                  "x-ms-enum": {
+                    "name": "AuthMode",
+                    "modelAsString": true
+                  },
+                  "readOnly": true,
+                  "description": "Authentication used for calling endpoint. Immutable and exclusive."
+                },
+                "uri": {
+                  "description": "The URI for sending inference requests.",
+                  "type": "string",
+                  "readOnly": true
+                }
+              },
+              "x-ms-discriminator-value": "InferenceEndpoint"
+            }
+          }
+        }
+      ]
+    },
+    "InferenceEndpointUpdateParameter": {
+      "description": "The parameters for updating an inference endpoint.",
+      "properties": { }
+    },
+    //////////////////////////////////////////////////////
+    // Inference Deployment
+    //////////////////////////////////////////////////////
+    "InferenceDeploymentResource": {
+      "type": "object",
+      "description": "Machine Learning Inference Deployment object wrapped into ARM resource envelope.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/Resource"
+        },
+        {
+          "type": "object",
+          "properties": {
+            "properties": {
+              "description": "Inference Deployment properties",
+              "properties": {
+                "models": {
+                  "description": "The list of model URIs.",
+                  "type": "array",
+                  "items": {
+                    "type": "string"
+                  }
+                },
+                // TODO: verify this is what autoscaler needs... also this seems to be in conflict
+                // with the sku in the base "Resource" class
+                "sku": {
+                  "$ref": "#/definitions/SkuDescription"
+                },
+                "weight": {
+                  "description": "weight of the deployment for load balancing.",
+                  "type": "integer"
+                },
+                "InferenceDeploymentProvisioningState": {
+                  // https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/Addendum.md#provisioningstate-property
+                  "type": "string",
+                  "enum": [
+                    "Unknown",
+                    "Scaling",
+                    "Creating",
+                    "Deleting",
+                    "Succeeded",
+                    "Failed",
+                    "Canceled"
+                  ],
+                  "x-ms-enum": {
+                    "name": "InferenceDeploymentProvisioningState",
+                    "modelAsString": true
+                  },
+                  "readOnly": true,
+                  "description": "The current provisioning state of the Inference Deployment resource."
+                },
+                "environmentConfigUri": {
```

Comment:
should we add a property to surface the containerImage being used?

Reply:
this leaves me with questions about what is expected in the Control Plane contract, since control plane is getting redesigned. I need to ask about this next meeting.